### PR TITLE
fix: Table interact-able columnTypes should not show in selectedRow

### LINKED ISSUE #15640: [Bug]: Table widget interactable columnTypes (button, icon button, menu button, and edit actions) should not be present in `selectedRow`, `triggeredRow`, and `updatedRow` properties
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Table widget interactable columnTypes (button, icon button, menu button, and edit actions) should not be present in `selectedRow`, `triggeredRow` properties.

### Steps To Reproduce

1. D&D a table widget.
2. add a custom column and change the type to button
3. Observe that the button column appears in `selectedRow` and `triggeredRow` properties.

### Public Sample App

_No response_

### Version

cloud

### PR DESCRIPTION
**Pull request by souma-ghosh**
## Description

Interact-able column types will not show in selectedRow, triggeredRow and updatedRow properties

Fixes #15640

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Jest tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/widgets/TableWidgetV2/widget/derived.js [hunk 1]
```diff
@@ -27,6 +27,24 @@ export default {
...
+              (primaryColumns?.[key]?.columnType
+                ?.toLowerCase()
+                ?.includes("button") ||
+                primaryColumns?.[key]?.columnType === "editActions") &&
+              primaryColumns[key]?.alias
+            ) {
+              return primaryColumns[key].alias;
+            }
+          })
+          .filter((alias) => alias)
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidgetV2/widget/derived.js:45]: 1. Do the filter before the map.
2. match the column type with the actual column value.
3. use Object values functions.
4. do indexOf search against an array.

**Review by sbalaji1192 — APPROVED** [review]

**Comment by souma-ghosh** [comment]
Table properties that won't show button columns and editActions data:

- selectedRow
- selectedRows
- updateRow
- updatedRows
- triggeredRow

**Comment by laveena-en** [comment]
Tested and verified.

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/TableV2/Inline_editing_spec.js [hunk 2]
```diff
@@ -10,174 +10,174 @@ describe("Table widget inline editing functionality", () => {
...
-        .contains(data.columnType)
-        .click();
-      cy.wait("@updateLayout");
-      cy.get(".t--property-pane-back-btn").click();
-      cy.get(`[data-rbd-draggable-id="step"] .t--card-checkbox input`).should(
-        data.expected,
-      );
-    });
-  });
+  // it("1. should check that edit check box is present in the columns list", () => {
```
Comment (sbalaji1192) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/Widgets/TableV2/Inline_editing_spec.js:13]: remove comment

**Review by sbalaji1192 — APPROVED** [review]

**Pull request merged**